test-framework/security-jwt/src/main/java/io/quarkus/test/security/jwt/JsonWebTokenProducer.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.test.security.jwt;
+
+import javax.enterprise.context.RequestScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import io.quarkus.arc.AlternativePriority;
+import io.quarkus.security.identity.SecurityIdentity;
+
+@RequestScoped
+@AlternativePriority(3)
```

Comment:
Does this mean that you can't mix actual JWT and @TestSecurity in the same module, as this will always take precedence?

Reply:
@stuartwdouglas Hi Stuart, yes. Both `smallrye-jwt` and `quarkus-oidc` ship their own `JsonWebToken` producers.

I don't have problems at all with not shipping this test producer if it were possible to support `@Inject JsonWebToken jwt;` without the producer but I'm not sure it is possible - but it is the endpoint code which has `@ Inject JsonWebToken jwt;` which users would like to test using `TestSecurity` so this injection has to be supported...

Reply:
But is it really important to have for ex `quarkus-smallrye-jwt` and `test-security-jwt` dependencies combined ? We already have Wiremock and Keycloak-specific test module support for the integration testing of `smallrye-jwt` and `quarkus-oidc` when they need to test the `real` producers of JWT. My understanding when someone is using `test-security-*` they don't need any other dependencies to test ?

Reply:
@stuartwdouglas let me experiment a bit more

Reply:
@stuartwdouglas `SecurityidentityAugmentor` is currently not supported with `TestSecurity` (may be they should not be as they can require the remote connections - or may be it can be done optionally).
But for now it works well with the existing `quarkus-smallrye-jwt` `JsonWebToken` producer. But not with `quarkus-oidc` - so I'll see tomorrow if I can fix it for the bearer tokens only (for the code flow we'd most likely need `test-security-oidc` as it also has id tokens, userinfo to take care about)